src/pyramid/request.py

```diff
@@ -330,3 +333,117 @@ def apply_request_extensions(request, extensions=None):
         InstancePropertyHelper.apply_properties(
             request, extensions.descriptors
         )
+
+
+class RequestLocalCache:
+    """
+    A store that caches values during for the lifecycle of a request.
+
+    Instantiate a cache object and use it to decorate methods or functions
+    that accept a request parameter. Any other arguments are not used as
+    cache keys so make sure they are constant across calls per-request.
+
+    Wrapping methods:
+
+    .. code-block:: python
+
+        class SecurityPolicy:
+            identity_cache = RequestLocalCache()
```

Comment:
Is is required that the cache instance lives in the class namespace?
Or would this work as well:
```python
class SecurityPolicy:
    @RequestLocalCache()
    def authenticated_identity(self, request):
        ...
```

Reply:
Yes your snippet works fine, the goal of the example is to show how to maintain a reference to the cache so you can modify it directly. For example if you wanted to change the value if `forget()` is called, you'd need to be able to reference the cache from there.